Automattic/jetpack
# Move JITM's REST API endpoints into the package

This PR moves the REST APIs endpoints used by the JITM into the package to make it independent from the Jetpack plugin

#### Changes proposed in this Pull Request:

* Move JITM's REST API endpoints into the package

#### Jetpack product discussion


1199702491434901-as-1199702491434934

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Checkout this PR
* Make sure to refresh the composer autoloader for the Jetpack plugin and the JITM package ( `composer dump-autoload` in their folders should do it)
* Visit your admin and make sure JITMs are still working
* Open the Network inspector
* Edit `Automattic\Jetpack\JITMS\Rest_Api_Endpoints::delete_jitm_message_permission_callback` and force it to return the `WP_Error`.
* On the browser, click to dismiss the JITM
* Check the debugger and make sure the error code and message are correct.
* Undo your changes to the file
* Dismiss the message and make sure it works

QUESTION:

I had to enable the JITM always, and not only in `is_admin` otherwise the REST endpoints would not be registered. I tried checking on the `REST_REQUEST` constant but apparently we are too early in the flow there. Do you think that's ok? Any ideas?

I thought about adding a `is_admin` check inside the package, but I guess once the package files were loaded the worse in terms of performance has already happened...
